### PR TITLE
feature/cvsb-9373

### PR DESCRIPTION
**Pull request by oalexandru**

### CODE REVIEW

File: src/services/TestResultsService.ts [hunk 1]
```diff
@@ -434,7 +434,7 @@ export class TestResultsService {
     let bool = false;
     if (payload.testTypes) {
       payload.testTypes.forEach((testType) => {
-        if ((this.isTestTypeAdr(testType) && testType.testResult === TEST_RESULT.PASS) && !testType.certificateNumber) {
+        if ((this.isTestTypeAdr(testType) && testType.testResult === TEST_RESULT.PASS) && payload.testStatus !== TEST_STATUS.CANCELLED && !testType.certificateNumber) {
```
Comment (KHodderDVSA) [inline src/services/TestResultsService.ts]: Unclear form story what this is doing. Can something be a testResult = pass AND cancelled?

File: tests/unit/insertTestResult.unitTest.ts [hunk 2]
```diff
@@ -1357,10 +1357,10 @@ describe("insertTestResult", () => {
         });
     });
 
-    context("when inserting a testResult that has an ADR testType without expiryDate", () => {
+    context("when inserting a testResult that has a pass ADR testType without expiryDate", () => {
         it("should throw 400 and descriptive error message", () => {
             const testResultWithAdrTestTypeWithoutExpiryDate = testResultsPostMock[6];
-            delete testResultWithAdrTestTypeWithoutExpiryDate.testTypes[0].testExpiryDate;
```
Comment (KHodderDVSA) [inline tests/unit/insertTestResult.unitTest.ts]: Will this scenario (no testExpiryDate entry at all, or testExpiryDate undefined) not actually occur?

File: tests/unit/insertTestResult.unitTest.ts [hunk 3]
```diff
@@ -1389,22 +1389,21 @@ describe("insertTestResult", () => {
...
         });
     });
 
-    context("when inserting a testResult that has an ADR testType without a certificateNumber", () => {
-        it("should throw 400 and descriptive error message", () => {
-            const testResultWithAdrTestTypeWithoutExpiryDate = testResultsPostMock[6];
-            delete testResultWithAdrTestTypeWithoutExpiryDate.testTypes[0].certificateNumber;
+    context("when inserting a testResult that has a fail ADR testType without expiryDate", () => {
+        it("should not throw error", () => {
+            const testResultWithAdrTestTypeWithoutExpiryDate = JSON.parse(JSON.stringify(testResultsPostMock[6]));
```
Comment (KHodderDVSA) [inline tests/unit/insertTestResult.unitTest.ts]: Better to use lodash or similar to do a clean cloneDeep. JSON.parse(JSON.stringify(X)) can have unintended side effects.

File: tests/unit/insertTestResult.unitTest.ts [hunk 4]
```diff
@@ -1432,6 +1431,47 @@ describe("insertTestResult", () => {
...
+                })
+                .catch((error: { statusCode: any; body: any; }) => {
+                    expect.fail();
+                });
+        });
+    });
+
+    context("when inserting a submitted testResult that has an ADR testType without a certificateNumber", () => {
+        it("should throw 400 and descriptive error message", () => {
+            const testResultWithAdrTestTypeWithoutCertificateNumber = testResultsPostMock[6];
```
Comment (KHodderDVSA) [inline tests/unit/insertTestResult.unitTest.ts]: Definitely don't mutate un-cloned instances of global objects. Will potentially be re-used in other tests. Lodash's cloneDeep is your friend here.

File: src/services/TestResultsService.ts (hunk 1 above)
Comment (KHodderDVSA) [reply src/services/TestResultsService.ts]: Apparently the answer is "yes"

File: tests/unit/insertTestResult.unitTest.ts (hunk 2 above)
Comment (KHodderDVSA) [reply tests/unit/insertTestResult.unitTest.ts]: Front end will always send a field of this name - null if no real  value

**Comment by JonLazariniDD** [comment]
outdated

**Pull request closed without merging**